cfme/tests/services/test_provision_stack.py

```diff
@@ -121,129 +72,148 @@ def stack_data(appliance, provider, provisioning):
+    if stack.exists:
+        stack.retire_stack()
+    if template.exists:
+        template.delete()
+
+
+@pytest.yield_fixture
+def catalog():
+    cat_name = "cat_{}".format(fauxfactory.gen_alphanumeric())
+    catalog = Catalog(name=cat_name, description="my catalog")
+    catalog.create()
+    yield catalog
+    if catalog.exists:
+        catalog.delete()
+
 
+@pytest.yield_fixture
+def catalog_item(dialog, catalog, template, provider, dialog_name):
+    item_name = fauxfactory.gen_alphanumeric()
+    catalog_item = CatalogItem(item_type="Orchestration",
+                               name=item_name,
+                               description="my catalog",
+                               display_in=True,
+                               catalog=catalog,
+                               dialog=dialog_name,
+                               orch_template=template,
+                               provider=provider)
+    catalog_item.create()
+    yield catalog_item
+    if catalog_item.exists:
+        catalog_item.delete()
+
+
+@pytest.fixture
+def order_service(appliance, catalog_item, stack_data):
     service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog,
                                        catalog_item.name, stack_data)
     service_catalogs.order()
-    logger.info('Waiting for cfme provision request for service {}'.format(catalog_item.name))
-    request_description = catalog_item.name
-    provision_request = appliance.collections.requests.instantiate(request_description,
+
+
+@pytest.yield_fixture
+def provision_request(appliance, catalog_item, order_service):
+    provision_request = appliance.collections.requests.instantiate(catalog_item.name,
                                                                    partial_check=True)
-    provision_request.wait_for_request()
-    assert provision_request.is_succeeded()
+    logger.info('Waiting for cfme provision request for service %s', catalog_item.name)
+    provision_request.wait_for_request(method='ui')
+    yield provision_request
+    if provision_request.exists:
+        provision_request.remove_request()
+
+
+@pytest.yield_fixture
+def service(appliance, provision_request):
+    last_message = provision_request.get_request_row_from_ui()['Last Message'].text
+    service_name = last_message.split()[2].strip('[]')
+    service = MyService(appliance, service_name)
+    yield service
+    if service.exists:
+        service.delete()
+
 
+@pytest.fixture
+def stack(appliance, provider, stack_data):
+    return appliance.collections.stacks.instantiate(stack_data['stack_name'], provider=provider)
 
-def test_reconfigure_service(appliance, provider, provisioning, catalog, catalog_item, request,
-                             stack_data):
+
+def test_provision_stack(provision_request, stack):
     """Tests stack provisioning
 
     Metadata:
         test_flag: provision
     """
-    catalog_item, template = catalog_item
+    assert provision_request.is_succeeded()
```

Comment:
**REQUIRED:** I'm not 100% sure about this. It seems that the majority of what happens here happens in fixtures. The test body should contain what is being tested. In this case the test body is only testing that the response was successful.

Reply:
Some of the tests below use the fixtures correctly but I believe this one is an oversimplification, probbaly so that the fixtures can be reused.

Reply:
@psav I don't think it's an oversimplification. The test does exactly what it describes. It checks if the stack was successfully provisioned and if the stack exists. Other things are not related to the test itself. We don't test service ordering or catalog item creation or template creation.

Reply:
I think we need someone else to come in and resolve this one @jkrocil perhaps
To me, the test is called ```test_provision_stack```. Whilst I agree that the catalog item creation and the template creation shouldn't be part of the test. I **do** think that the act of ordering the stack **is** something that should form part of the test. In reality it could be that this is the **only** place we should ever see this inside a test. I would be perfectly happy for anything that use a stack in the test to provision it as part of a fixture. However I do firmly believe that this should be part of the test in this case. I am of course sometimes wrong, may-hap often. So I will let @jkrocil give the final decision on it.